src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          delete stateCopy[keys];
+        }
+        break;
 
+      default:
+        stateCopy = {};
+    }
```

Comment:
The default case in the switch statement should not clear the state. Instead, it should handle unexpected action types, possibly by throwing an error or logging a warning. This ensures that the function behaves predictably and helps to identify issues when an incorrect action type is used.